vhost-user-backend/src/handler.rs

```diff
@@ -257,6 +257,19 @@ where
         Ok(())
     }
 
+    fn reset_device(&mut self) -> VhostUserResult<()> {
+        // Disable all vrings
```

Comment:
Should we check if `VHOST_USER_PROTOCOL_F_RESET_DEVICE` is negotiated or at least exposed by the backend?

The spec says:
```
``VHOST_USER_RESET_DEVICE``
...
  Only valid if the ``VHOST_USER_PROTOCOL_F_RESET_DEVICE`` protocol
  feature is set by the back-end.
```

Reply:
You mean adding a `check_protocol_feature()` function like `check_feature()`, and then run `self.check_protocol_feature(VhostUserProtocolFeatures::RESET_DEVICE)?;`? I don’t think other handlers do that, yet (e.g. `get_config()` should check `VHOST_USER_PROTOCOL_F_CONFIG`, but sure.

Reply:
Ah, `check_proto_feature()` it is, that’s how `FrontendInternal` calls it.

Reply:
Oh, I was checking `get_config()` and I noticed that we do the check in `BackendReqHandler::handle_request()` so now I'm thinking if it makes more sense to do it there, where we already have `check_proto_feature`.

Sorry for the confusion :-(

Reply:
That said, IMHO is better what we are doing here, so not sure if in the future we should move all the checks here.

So I'm fine with both this version or just move the check in the handle_request(). up to you ;-)

Reply:
Oh, right! I’ll move it there.

(The only other thing that checks feature availability in `impl VhostUserBackendReqHandlerMut for VhostUserHandler` is `fn set_vring_enable()`, which… also does it in `handle_request()`. Double-checking isn’t bad by any means, but that does show everyone expects there to be a check in `handle_request()`, and I think it makes sense to follow the majority of implementations and only check there then.)